tests/func/experiments/test_checkpoints.py

```diff
@@ -188,3 +195,90 @@ def test_resume_non_head_checkpoint(
     )
     new_head = first(results)
     assert orig_branch != dvc.experiments.get_branch_by_rev(new_head)
+
+
+@pytest.fixture
+def clear_env(monkeypatch):
+    yield
+    monkeypatch.delenv(DVC_EXP_GIT_REMOTE, raising=False)
+    monkeypatch.delenv(DVC_EXP_AUTO_PUSH, raising=False)
+
+
+@pytest.mark.parametrize("use_url", [True, False])
+def test_auto_push_during_iterations(
+    tmp_dir,
+    scm,
+    dvc,
+    checkpoint_stage,
+    git_upstream,
+    local_remote,
+    use_url,
+    monkeypatch,
+    mocker,
+    clear_env,
+):
+    # set up remote repo
+    remote = git_upstream.url if use_url else git_upstream.remote
+    git_upstream.scm.fetch_refspecs(str(tmp_dir), ["master:master"])
+    monkeypatch.setenv(DVC_EXP_GIT_REMOTE, remote)
+    auto_push_spy = mocker.spy(BaseExecutor, "_auto_push")
+
+    # without auto push
+    results = dvc.experiments.run(checkpoint_stage.addressing)
+    assert auto_push_spy.call_count == 0
+
+    # add auto push
+    monkeypatch.setenv(DVC_EXP_AUTO_PUSH, "true")
+    results = dvc.experiments.run(checkpoint_stage.addressing)
+    assert (tmp_dir / "foo").read_text() == "4"
```

Comment:
not complaining, but this reminds me of
![xkcd#221](https://imgs.xkcd.com/comics/random_number.png)

Reply:
Your comment reminds me of this old blogpost: https://testing.googleblog.com/2014/07/testing-on-toilet-dont-put-logic-in.html.

Reply:
Like Casper said it's not a big deal, but we could actually get rid of these magic numbers since it comes from a defined tests constant:
https://github.com/iterative/dvc/blob/fb96b13f13948d775a3061c03868db3775f6c665/tests/func/experiments/conftest.py#L7

the `4` is the number of `exp run` calls multiplied by the number of iterations per `exp run` so it would be a bit clearer to write it as `2 * DEFAULT_ITERATIONS`